app/MMB/src/scenes/tripServices/transportation/TransportationMap.js

```diff
@@ -0,0 +1,350 @@
+    this.state = {
+      region: {
+        latitude: props.params.location.lat,
+        longitude: props.params.location.lng,
+        latitudeDelta: 0.5,
+        longitudeDelta: 0.5,
+      },
+      markers: {
+        markerA: null,
+        markerB: null,
+      },
+      destination: '',
+      showUserLocation: false,
+      currentLocation: null,
+    };
+  }
+
+  static navigationOptions = (props: Props) => {
+    function goBack() {
+      props.navigation.goBack(null);
+    }
+
+    return {
+      headerLeft: (
+        <HeaderBackButton
+          tintColor={defaultTokens.paletteProductNormal}
+          onPress={goBack}
+        />
+      ),
+      headerTitle: (
+        <HeaderTitle>
+          <Translation id="mmb.trip_services.transportation.map.title" />
+        </HeaderTitle>
+      ),
+      headerRight: (
+        <HeaderButton.Right onPress={props.openLink} disabled={props.disabled}>
+          <HeaderButton.Text>
+            <Translation id="mmb.trip_services.transportation.map.right_button" />
+          </HeaderButton.Text>
+        </HeaderButton.Right>
+      ),
+      headerStyle: {
+        backgroundColor: defaultTokens.paletteWhite,
+        borderBottomWidth: 0,
+      },
+    };
+  };
+
+  componentDidMount() {
+    this.props.navigation.setParams({
+      disabled: true,
+      openLink: this.openLink,
+    });
+  }
+
+  componentDidUpdate() {
+    const { navigation } = this.props;
+    const { markerA, markerB } = this.state.markers;
+    if (
+      markerA != null &&
+      markerB != null &&
+      navigation.state.params.disabled
+    ) {
+      navigation.setParams({ disabled: false });
+    }
+  }
+
+  openLocationPicker = async () => {
+    const { currentLocation } = this.state;
+    let currentAddress;
+
+    if (currentLocation != null) {
+      const { latitude, longitude } = currentLocation;
+      currentAddress = await this.getFormattedAddress({
+        latitude,
+        longitude,
+      });
+    }
+
+    this.props.navigation.navigate('AddressPickerScreen', {
+      currentAddress,
+    });
+  };
+
+  openSettings = () => {
+    Linking.openURL('app-settings:');
+  };
+
+  alertOpenSettings = () => {
+    Alert.translatedAlert(
+      undefined,
+      {
+        id: 'mmb.trip_services.transportation.map.current_position_alert',
+      },
+      [
+        {
+          text: { id: 'mmb.alert.button.ok' },
+          undefined,
+          style: 'default',
+        },
+        {
+          text: { id: 'mmb.alert.button.settings' },
+          onPress: this.openSettings,
+          style: 'default',
+        },
+      ],
+    );
+  };
+
+  onRegionChange = (region: Region) => {
+    this.setState({ region });
+  };
+
+  buildWhitelabelURL = (whitelabelURL: string) => {
+    const { markerA, markerB } = this.state.markers;
+    if (markerA != null && markerB != null) {
+      const URLbase = whitelabelURL.replace(/\?.*/, '');
+      const URLparameters = whitelabelURL.replace(/.*\?/, '');
```

Comment:
maybe we should rather just change how we return the whitelabel url from the graphql?

Reply:
Like you send the arguments, and it gives you back the correct URL?

Reply:
I mean just send the basic, here it is being picked apart and put back together. lets just send `https://url.withouth.parameters/`